test/unit/queue/src/QueueTest.cpp

```diff
@@ -65,7 +43,13 @@ TEMPLATE_LIST_TEST_CASE("queueCallbackIsWorking", "[queue]", TestQueues)
 
     alpaka::enqueue(f.m_queue, [&]() { promise.set_value(true); });
 
-    LOOPED_CHECK(30, 100, promise.get_future().get());
+    // A non-blocking queue will always stay empty because the task has been executed immediately.
+    if(!alpaka::test::IsBlockingQueue<typename Fixture::Queue>::value)
+    {
+        alpaka::wait(f.m_queue);
+    }
+
+    CHECK(promise.get_future().get());
```

Comment:
No wait should be necessary, since the `std::future::get()` blocks until the promise is satisfied, and thus synchronizes with the task.

Reply:
yes you are right